File: backend/src/main/java/isep/webapp/terasync/model/user/DeliveryService.java
package isep.webapp.terasync.model.user;

import isep.webapp.terasync.model.odSystem.CustomerOrder;
import isep.webapp.terasync.model.odSystem.DeliveryServiceList;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.util.List;

@Entity
@Setter
@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class DeliveryService extends User {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "deliveryServiceId")
    private int deliveryServiceId;
    @Column(name = "name")
    private String name;
    @Column(name = "email")
    private String email;
    @Column(name = "username")
    private String username;
    @Column(name = "password")
    private String password;
    @Column(name = "token")
    private String token;
    @CreationTimestamp
    @Column(name = "registrationDate")
    private String registrationDate;
    @Column(name = "emailVerified")
    private boolean emailVerified;
    @Column(name= "approved")
    private boolean approved;
    @Column(name = "pfpImgPath")
    private String pfpImgPath;

    @OneToMany(
            fetch = FetchType.EAGER,
            targetEntity = CustomerOrder.class,
            cascade = CascadeType.ALL
    )
    @JoinColumn(name = "deliveryServiceId")
    private List<CustomerOrder> customerOrders;

    @OneToMany(
            fetch = FetchType.EAGER,
            targetEntity = DeliveryServiceList.class,
            cascade = CascadeType.ALL
    )
    @JoinColumn(name = "deliveryServiceId")
    private List<DeliveryServiceList> deliveryServiceLists;

    @OneToMany(
            fetch = FetchType.EAGER,
            targetEntity = DeliveryPerson.class,
            cascade = CascadeType.ALL
    )
    @JoinColumn(name = "deliveryServiceId")
    private List<DeliveryPerson> deliveryPeople;

    @Override
    public int getUserId() {
        return deliveryServiceId;
    }

    @Override
    public boolean getEmailVerified() {
        return emailVerified;
    }
}

File: front/src/app/model/user/delivery.service.ts
import {User} from "./user";
import {DeliveryServiceList} from "../odSystem/delivery.service.list";
import {DeliveryPerson} from "./delivery.person";
import {CustomerOrder} from "../odSystem/customer.order";

export class DeliveryService extends User {
  override deliveryServiceId: number | undefined
  override approved: boolean | undefined;

  override deliveryServiceLists: DeliveryServiceList[] = [];
  override customerOrders: CustomerOrder[] = [];
  override deliveryPeople: DeliveryPerson[] = [];

  constructor(name: string, email: string, username: string,
              password: string, deliveryServiceId?: number | undefined,
              registrationDate?: string | undefined,  token?: string | undefined,
              emailVerified?: boolean | undefined, approved?: boolean | undefined,
              pfpImgPath?: string | undefined) {
    super(email, username, password, deliveryServiceId, registrationDate, token, emailVerified, pfpImgPath, name);
    this.deliveryServiceId = deliveryServiceId;
    this.approved = approved;
  }

  static override fromJson(jsonDeliveryService: DeliveryService) {
    let deliveryService = new DeliveryService(jsonDeliveryService.name!, jsonDeliveryService.email, jsonDeliveryService.username,
      jsonDeliveryService.password, jsonDeliveryService.deliveryServiceId, jsonDeliveryService.registrationDate,
      jsonDeliveryService.token, jsonDeliveryService.emailVerified, jsonDeliveryService.approved, jsonDeliveryService.pfpImgPath)

    deliveryService.deliveryServiceLists = DeliveryServiceList.initializeDeliveryServiceLists(jsonDeliveryService);
    deliveryService.customerOrders = CustomerOrder.initializeCustomerOrders(jsonDeliveryService);
    deliveryService.deliveryPeople = DeliveryPerson.initializeDeliveryPeople(jsonDeliveryService);

    return deliveryService;
  }
}
